proposals/function-pointers.md

```diff
@@ -63,174 +64,173 @@ unsafe class Example {
 }
 ```
 
-A `func*` type is a pointer type which means it has all of the capabilities and restrictions of a standard pointer
+A `delegate*` type is a pointer type which means it has all of the capabilities and restrictions of a standard pointer
 type:
+
 - Only valid in an `unsafe` context.
-- Methods which contain a `func*` parameter or return type can only be called from an `unsafe` context.
+- Methods which contain a `delegate*` parameter or return type can only be called from an `unsafe` context.
 - Cannot be converted to `object`.
 - Cannot be used as a generic argument.
-- Can implicitly convert `func*` to `void*`.
-- Can explicitly convert from `void*` to `func*`.
+- Can implicitly convert `delegate*` to `void*`.
+- Can explicitly convert from `void*` to `delegate*`.
 
 Restrictions:
-- Custom attributes cannot be applied to a `func*` or any of its elements.
-- A `func*` parameter cannot be marked as `params`
-- A `func*` type has all of the restrictions of a normal pointer type.
+
+- Custom attributes cannot be applied to a `delegate*` or any of its elements.
```

Comment:
Custom attributes can't be applied today but we will likely want the ability to expose them in the future (to support `GCSuppressTransition`, other calling convention designations, etc).

Presumably, the compiler is currently planning on emitting these types directly as `FNPTR` type signatures. However, the runtime doesn't currently have any way to attach attributes to such type signatures.

Has this been taken into consideration since that would make it a breaking change to go from the `inline function pointer` syntax (the only thing available today) to change to some syntax that allows attributes?

CC. @333fred, @gafter, @jkotas

Reply:
> Custom attributes can't be applied today but we will likely want the ability to expose them in the future 

Is there a design stating how such support would be implemented in the runtime?

> Has this been taken into consideration since that would make it a breaking change to go from the inline function pointer syntax (the only thing available today) to change to some syntax that allows attributes?

No it hasn't because the underlying runtime support for this does not exist and would likely require changes to `calli` or the method in which the signature is provided to `calli`. Can't really take it into consideration if we don't know how the feature would be implemented in the runtime.

Reply:
Asked someone from the runtime team to comment on the relevant issue: <https://github.com/dotnet/roslyn/issues/39865#issuecomment-596800197>